Updates/0.3.3.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Plugins\QueuedTracking;
+
+use Piwik\Container\StaticContainer;
+use Piwik\Plugins\QueuedTracking\Queue\Manager;
+use Piwik\Updater;
+use Piwik\Updates;
+
+class Updates_0_3_3 extends Updates
+{
+    /**
+     * @var Manager
+     */
+    protected $queueManager;
+
+    public function doUpdate(Updater $updater)
+    {
+        $settings = $this->getSettings();
+        $numQueueWorkers = $settings->numQueueWorkers->getValue();
+
+        if ($numQueueWorkers <= 1) {
+            return; // it is pointless to rewrite one queue to the same queue
+        }
+
+        $this->rewriteQueues($numQueueWorkers);
```

Comment:
I see `moveSomeQueuesIfNeeded` does not do quite the same or may even have problems. So finding a good solution would be good and then moving it into a class so it can be re-used. It should not load all requests into memory as this will easily break